### PR TITLE
Implemented Component Item controller and database table

### PR DESCRIPTION
**Pull request by DrMadhushan**
* Added table for component item
* Added seed data (2 items)
* Updated the controller file for comp item
* In ComponentItemController.php not included the componentType foriegn key. (just added as attributes)
* Schema (not fully implemented)
- https://drive.google.com/file/d/1hhZXSwS1GnxY2jNdhWOYPWFYrUkefZVT/view?usp=sharing

### CODE REVIEW

**Comment by NuwanJ** [comment]
Changed the target branch of the PR from *'main'* into *'issue_029'*. You can merge any of your implementations related to issue #29 into this branch, without review or approval from others.

We will review the implementation when merging a PR from branch issue_029 to main or dev.

File: app/Http/Controllers/Backend/ComponentItemController.php [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\View\Factory|\Illuminate\Contracts\View\View
+     */
+    public function create()
+    {
+        $types = ComponentType::pluck('title', 'id');
```
Comment (NuwanJ) [inline app/Http/Controllers/Backend/ComponentItemController.php:33]: The title should be changed into 'name' in this case

File: database/migrations/2021_07_28_190230_create_component_items_table.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            $table->string("productCode")->nullable();
+
+            $table->text("specifications")->nullable();
+            $table->text("description")->nullable();
+
+            $table->boolean("isAvailable")->default(false);
+            $table->float("price")->nullable(); // in LKR
+
+            // Physical size in terms of appearance [small, medium, large] kind of
+            $table->string("size")->nullable()->default('');
```
Comment (NuwanJ) [inline database/migrations/2021_07_28_190230_create_component_items_table.php:31]: Are there any numerical parameters to define the size?

Laravel supports enum as a data type >  https://laravel.com/docs/8.x/migrations#column-method-enum

File: database/migrations/2021_07_28_190230_create_component_items_table.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+            // Thought to have type and family as a property rather than having different tables
+            $table->string("type")->nullable()->default('');
+            $table->string("family")->nullable()->default('');
```
Comment (NuwanJ) [inline database/migrations/2021_07_28_190230_create_component_items_table.php:35]: If nullable is used, better if there isn't any default value. (then the default = NULL)

File: database/migrations/2021_07_28_190230_create_component_items_table.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+            $table->string("name");
+            $table->string("brand")->nullable();
+            $table->string("productCode")->nullable();
+
+            $table->text("specifications")->nullable();
+            $table->text("description")->nullable();
+
+            $table->boolean("isAvailable")->default(false);
+            $table->float("price")->nullable(); // in LKR
```
Comment (NuwanJ) [inline database/migrations/2021_07_28_190230_create_component_items_table.php:28]: Shall we remove 'price' from here, and add it in the 'stocks' table as you previously suggested in your EER diagram?

Comment (DrMadhushan) [reply database/migrations/2021_07_28_190230_create_component_items_table.php:28]: Price in the stock table? will it make sense?
Bcz it's a property of the item right?

File: app/Http/Controllers/Backend/ComponentItemController.php (hunk 1 above)
Comment (DrMadhushan) [reply app/Http/Controllers/Backend/ComponentItemController.php]: changed

File: database/migrations/2021_07_28_190230_create_component_items_table.php (hunk 4 above)
Comment (NuwanJ) [reply database/migrations/2021_07_28_190230_create_component_items_table.php:28]: But for the same item, the price can be changed from one batch of purchases to another batch of purchases.

Ex: I can buy 5 Arduino UNO boards under Rs.800 now, and put them into the stocks while existing 3 Arduino Boards in the stocks which were purchased for Rs.850.

**Pull request merged**